wrappers/wrapper.rb

```diff
@@ -572,6 +573,88 @@ def detect_unfeasible_services(vrp)
       unfeasible
     end
 
+    def sequence_relation_with_no_reachable_vehicle?(unfeasible, vrp, relation)
+      return false unless Models::Relation::POSITION_TYPES.include?(relation.type)
+
+      vrp.vehicles.each{ |v|
+        unreachable_service_id = last_service_reachable_sequence(vrp, v, relation.linked_services)
+        next unless unreachable_service_id
+
+        service_met = false
+        relation.linked_services.each{ |service|
+          service_met = true if service.id == unreachable_service_id
+          next unless service_met
+
+          service.vehicle_compatibility[v.id] = false
+        }
+      }
+
+      previous_removed = false
+      to_delete_services = []
+      relation.linked_services.each{ |service|
+        previous_removed ||= vrp.vehicles.none?{ |vehicle|
+          service.vehicle_compatibility[vehicle.id]
+        }
+
+        if previous_removed
+          if Models::Relation::ALL_OR_NONE_RELATIONS.include?(relation.type)
+            to_delete_services = relation.linked_services
+            break false
+          else
+            to_delete_services << service
+          end
+        end
+      }
+
+      relation.linked_services -= to_delete_services
+
+      to_delete_services.each{ |service|
+        add_unassigned(unfeasible, vrp, service, "Service belongs to a relation of type #{relation.type} which makes it infeasible")
+      }
+      relation.linked_services.empty?
+    end
+
+    def last_service_reachable_sequence(vrp, vehicle, services)
+      vehicle_timewindow = vehicle.timewindow || Models::Timewindow.new(start: 0)
+      vehicle_duration = [vehicle.duration, vehicle_timewindow.safe_end(vehicle.cost_late_multiplier&.positive?) - vehicle_timewindow.start].compact.min
+      successive_activities = services.map{ |service|
+        [service.id, service.activity && [service.activity] || service.activities]
+      }
+      return nil if successive_activities.all?{ |_id, acts| acts.any?{ |a| a.timewindows.none?(&:end) } }
+
+      matrix = vrp.matrices.find{ |m| m.id == vehicle.matrix_id }
+      successive_activities = services.map{ |service|
+        [service.id, service.activity && [service.activity] || service.activities]
+      }
```

Comment:
Duplicate code
```      
      successive_activities = services.map{ |service|
        [service.id, service.activity && [service.activity] || service.activities]
      }
```

Reply:
But also are we sure that this is the most elegant way to do this?

Reply:
The other solution is to entirely migrate activity to activities